kimchi/src/circuits/polynomials/keccak.rs

```diff
@@ -0,0 +1,81 @@
+//! Keccak gadget
+use std::array;
+
+use ark_ff::PrimeField;
+
+use crate::circuits::{
+    gate::{CircuitGate, Connect},
+    polynomial::COLUMNS,
+    polynomials::{generic::GenericGateSpec, rot::create_witness_rot},
+    wires::Wire,
+};
+
+/// Creates the 5x5 table of rotation bits for Keccak modulo 64
+/// | x \ y |  0 |  1 |  2 |  3 |  4 |
+/// | ----- | -- | -- | -- | -- | -- |
+/// | 0     |  0 | 36 |  3 | 41 | 18 |
+/// | 1     |  1 | 44 | 10 | 45 |  2 |
+/// | 2     | 62 |  6 | 43 | 15 | 61 |
+/// | 3     | 28 | 55 | 25 | 21 | 56 |
+/// | 4     | 27 | 20 | 39 |  8 | 14 |
+pub const ROT_TAB: [[u32; 5]; 5] = [
+    [0, 36, 3, 41, 18],
+    [1, 44, 10, 45, 2],
+    [62, 6, 43, 15, 61],
+    [28, 55, 25, 21, 56],
+    [27, 20, 39, 8, 14],
+];
+
+impl<F: PrimeField> CircuitGate<F> {
+    /// Creates Keccak gadget.
+    /// Right now it only creates an initial generic gate with all zeros starting on `new_row` and then
+    /// calls the Keccak rotation gadget
+    pub fn create_keccak(new_row: usize) -> (usize, Vec<Self>) {
```

Comment:
This is just a toy function, it does not create a keccak gadget whatsoever, just does what the comment above says. What I am saying is that this module has limited functionality as it is super incomplete at this moment, but is useful to try out the Keccak rotation table. It will be an iteratively built gadget in the future. It feels bit unnecessary at this point to have an extend version of it, specially because Perhaps a full keccak may not even fit in one single circuit. Nonetheless, as we keep writing these gadgets and see how many rows they need, then yes we will have an extend version for it.

Reply:
Also, this is the rotation PR, for that reason I am just addressing ROT-related stuff